### PR TITLE
enh(dashboard): add more filters to host / service statuses count endpoints

### PR DESCRIPTION
**Pull request by adr-mo**
This PR fully rewrites the endpoints to get the statuses of hosts and services.
Why: because it is really specific to hosts and services and that we should not try to use the ResourceRepository to fit those use cases.

✅ PHPStan level max
✅ php-cs-fixer
✅ Unit tests added for both endpoints

image

API contract unchanged. Example for services

image

**Fixes** # (issue)

## Type of change

- [ ] Patch fixing an issue (non-breaking change)
- [ ] New functionality (non-breaking change)
- [ ] Breaking change (patch or feature) that might cause side effects breaking part of the Software

## Target serie

- [ ] 22.10.x
- [ ] 23.04.x
- [ ] 23.10.x
- [ ] 24.04.x
- [ ] master

<h2> How this pull request can be tested ? </h2>

Please describe the **procedure** to verify that the goal of the PR is matched. Provide clear instructions so that it can be **correctly tested**.

Any **relevant details** of the configuration to perform the test should be added.

#### Community contributors & Centreon team

- [ ] I have followed the **coding style guidelines** provided by Centreon
- [ ] I have commented my code, especially new **classes**, **functions** or any **legacy code** modified. (***docblock***)
- [ ] I have commented my code, especially **hard-to-understand areas** of the PR.
- [ ] I have **rebased** my development branch on the base branch (master, maintenance).

### CODE REVIEW

File: centreon/doc/API/latest/onPremise/Realtime/Services/FindServicesStatus.yaml [hunk 1]
```diff
@@ -1,19 +1,24 @@
 get:
   tags:
-    - Resource
-  summary: "Count hosts by status"
+    - Service
+  summary: "Count services by status"
   description: |
-    Count hosts by their status (UP/DOWN/UNREACHABLE/PENDING)
+    Count hosts by their status (OK/WARNING/CRITICAL/UNKNOWN/PENDING)
```
Comment (jeremyjaouen) [inline centreon/doc/API/latest/onPremise/Realtime/Services/FindServicesStatus.yaml:6]:
```suggestion
    Count services by their status (OK/WARNING/CRITICAL/UNKNOWN/PENDING)
```

**Review by jeremyjaouen — DISMISSED** [review]

**Review by wtermellil — APPROVED** [review]

**Review by jdelpierre — DISMISSED** [review]
I approve only the test E2E changes

**Review by jeremyjaouen — DISMISSED** [review]

File: centreon/lang/fr_FR.UTF-8/LC_MESSAGES/messages.po [hunk 2]
```diff
@@ -18978,6 +18978,12 @@ msgstr "Impossible de migrer les mots de passe"
 msgid "No vault configured"
 msgstr "Aucun coffre configuré"
 
+msgid "You are not allowed to access real time hosts"
+msgstr "Vous n'êtes pas autorisé à accéder aux hôtes dans le temps réel"
```
Comment (adr-mo) [inline centreon/lang/fr_FR.UTF-8/LC_MESSAGES/messages.po:18982]:
```suggestion
msgstr "Vous n'êtes pas autorisé à accéder aux hôtes dans le cadre du temps réel"
```

File: centreon/lang/fr_FR.UTF-8/LC_MESSAGES/messages.po [hunk 3]
```diff
@@ -18978,6 +18978,12 @@ msgstr "Impossible de migrer les mots de passe"
...
+
+msgid "You are not allowed to access real time services"
+msgstr "Vous n'êtes pas autorisé à accéder aux services dans le temps réel"
```
Comment (adr-mo) [inline centreon/lang/fr_FR.UTF-8/LC_MESSAGES/messages.po:18985]:
```suggestion
msgstr "Vous n'êtes pas autorisé à accéder aux services dans le cadre du temps réel"
```

**Review by jeremyjaouen — APPROVED** [review]

**Review by smau-centreon — APPROVED** [review]

**Review by wtermellil — APPROVED** [review]

**Review by jdelpierre — APPROVED** [review]

**Pull request merged**